lib/zipper.mli

```diff
@@ -0,0 +1,18 @@
+type 'a t = {
```

Comment:
I believe resolved...?

Reply:
Not really :)

Let me try to explain: in OCaml, there is a clear distinction between the interface (or in other words, the API of a module) and its implementation. The former is in the `.mli`, and the latter in the `.ml`.

The API should contain everything you need to _use_ a module, but nothing more. In particular, in a `.mli` you have the name of a value, and it's type, but _how_ the value is implemented is hidden.
Having this separation is very nice, because it allows to change easily the implementation, without breaking any user. But for this to really work we need to not only hide the implementation of values, but also the one of types!
OCaml allows to define "abstract types": types whose definition is hidden:
```ocaml
(** dict.ml *)
type t = (int * string) list
let find dict key = ...

(** dict.mli *)
type t
val find : t -> int -> string
```

If we included the definition of the type `t` in the interface, some users could want to write code that depend on the fact that a dictionary is an association list, which is an implementation detail. And, if we want to change that to something else, say a Hashtable, then this would break all code relying on the implementation detail.

Since interface are the only thing that should be looked at when using a module, documentation comments should go there, and not in the `.ml`! For instance, the docs in ocaml.irg (such as [this one](https://ocaml.org/p/bos/latest/doc/Bos/OS/File/index.html)) are generated from `.mli`s that contain doc comments!